include/deal.II/lac/trilinos_tpetra_sparsity_pattern.h

```diff
@@ -0,0 +1,1378 @@
+       */
+      size_type
+      bandwidth() const;
+
+      /**
+       * Return whether the object is empty. It is empty if no memory is
+       * allocated, which is the same as when both dimensions are zero.
+       */
+      bool
+      empty() const;
+
+      /**
+       * Return whether the index (<i>i,j</i>) exists in the sparsity pattern
+       * (i.e., it may be nonzero) or not.
+       */
+      bool
+      exists(const size_type i, const size_type j) const;
+
+      /**
+       * Return whether a given @p row is stored in the current object
+       * on this process.
+       */
+      bool
+      row_is_stored_locally(const size_type i) const;
+
+      /**
+       * Determine an estimate for the memory consumption (in bytes) of this
+       * object. Currently not implemented for this class.
+       */
+      std::size_t
+      memory_consumption() const;
+
+      /** @} */
+      /**
+       * @name Adding entries
+       */
+      /** @{ */
+      /**
+       * Add the element (<i>i,j</i>) to the sparsity pattern.
+       */
+      void
+      add(const size_type i, const size_type j);
+
+
+      /**
+       * Add several elements in one row to the sparsity pattern.
+       */
+      template <typename ForwardIterator>
+      void
+      add_entries(const size_type row,
+                  ForwardIterator begin,
+                  ForwardIterator end,
+                  const bool      indices_are_sorted = false);
+
+      virtual void
+      add_row_entries(
+        const dealii::types::global_dof_index                  &row,
+        const ArrayView<const dealii::types::global_dof_index> &columns,
+        const bool indices_are_sorted = false) override;
+
+      using SparsityPatternBase::add_entries;
+
+      /** @} */
+      /**
+       * @name Access of underlying Trilinos data
+       */
+      /** @{ */
+
+      /**
+       * Return a Teuchos::RCP to the underlying Trilinos Tpetra::CrsGraph
+       * data that stores the sparsity pattern.
+       */
+      Teuchos::RCP<GraphType>
+      trilinos_sparsity_pattern() const;
+
+      /**
+       * Return a const Teuchos::RCP to the underlying Trilinos Tpetra::Map that
+       * sets the parallel partitioning of the domain space of this sparsity
+       * pattern, i.e., the partitioning of the vectors matrices based on this
+       * sparsity pattern are multiplied with.
+       */
+      Teuchos::RCP<const MapType>
+      domain_partitioner() const;
+
+      /**
+       * Return a const Teuchos::RCP to the underlying Trilinos Tpetra::Map that
+       * sets the partitioning of the range space of this sparsity pattern,
+       * i.e., the partitioning of the vectors that are result from matrix-
+       * vector products.
+       */
+      Teuchos::RCP<const MapType>
+      range_partitioner() const;
+
+      /**
+       * Return the underlying MPI communicator.
+       */
+      Teuchos::RCP<const Teuchos::Comm<int>>
+      get_mpi_communicator() const;
```

Comment:
I will adapt that function accordingly. Since the `Teuchos::RCP<const Teuchos::Comm<int>>` is frequently needed (at least internally), I will also add a new function:
```
Teuchos::RCP<const Teuchos::Comm<int>>
      get_teuchos_mpi_communicator() const;
```

Reply:
Yes, that seems like the right approach!

Reply:
Done!